x-pack/plugins/observability_solution/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -126,6 +142,14 @@ export const createMetricThresholdExecutor =
     const { criteria } = params;
     if (criteria.length === 0) throw new Error('Cannot execute an alert with 0 conditions');
 
+    const groupBy = castArray<string>(params.groupBy);
+    const assetTypeByAssetId = Object.values(SupportedAssetTypes).reduce((acc, curr) => {
+      acc[findInventoryModel(curr).fields.id] = curr;
+      return acc;
+    }, {} as Record<string, InventoryItemType>);
+    const supportedAssetId = groupBy.find((field) => !!assetTypeByAssetId[field]);
+    const nodeType = supportedAssetId ? assetTypeByAssetId[supportedAssetId] : undefined;
```

Comment:
Prolly needs to add a comment here too. This shenanigan here is for seeing if the alert was created with a groupby field that the asset details supports.